qiskit/primitives/containers/options.py

```diff
@@ -0,0 +1,41 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Options class
+"""
+
+from __future__ import annotations
+
+from abc import ABC
+from typing import Union
+
+from .dataclasses import mutable_dataclass
+
+
+@mutable_dataclass
+class BasePrimitiveOptions(ABC):
+    """Base class of options for primitives."""
+
+    def update(self, options: BasePrimitiveOptions | None = None, **kwargs):
```

Comment:
I would expect this to have a signature like `dict.update` but extended to also support `BasePrimitiveOptions` along with `Mapping` for the first arg.
```suggestion
     def update(self, options: BasePrimitiveOptions | Mapping | None = None, **kwargs):
```
so a single field can s

Reply:
Thanks. I updated.